### DISCUSSION TITLE
Question: Converting from Esri or System.Spatial Geometry to WKB

### DISCUSSION THREAD

**Discussion by nlraley** [NetTopologySuite/NetTopologySuite#585] [Q&A]
I am trying to figure out how one would convert either an Esri or System.Spatial Geometry class to the NetTopologySuite Geometry class so that I can write the WKB of the Geometry.  I am unable to find any documentation on how to do this.  Is this possible?

**Comment by DGuidi** [comment]
you need to convert the geometry by yourself, I think...
See the [Getting Started](https://github.com/NetTopologySuite/NetTopologySuite/wiki/GettingStarted) section.

Maybe this [old sharpmap provider](https://github.com/SharpMap/SharpMap/blob/756f9925aaa1b04196521218041db247bb02b263/SharpMap.Extensions/Data/Providers/NtsGeometryConverter.cs) can help you

**Comment by JeffJacobson** [comment]
I had trouble with SQL Spatial Geometry. I modified my SQL query to use [AsBinaryZM](https://docs.microsoft.com/en-us/sql/t-sql/spatial-geometry/asbinaryzm-geometry-datatype) method on the Geometry column value, then...

```csharp
using DbDataReader? reader = command.ExecuteReader();

while (reader.Read())
{
    // Read the binary geometry into a NetTopologySuite Geometry object.
    System.IO.Stream? shapeStream = reader.GetStream(2); // 2 is just the column ID telling which row has the geometry column.
    Geometry? geometry = wkbReader.Read(shapeStream); // Use WKBReader to convert the stream to NTS geometry object.
}
```

**Reply by DGuidi** [reply]
what kind of trouble do you have? you receive some error or exception? 
please add also the code you use to create the `wkbReader` instance
